spring-rabbit/src/test/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainerTests.java

```diff
@@ -564,6 +567,24 @@ public Object answer(InvocationOnMock invocation) throws Throwable {
 		container.stop();
 	}
 
+	@Test
+	public void testContainerNotRecoveredAfterExhaustingRecoveryBackOff() throws Exception {
+		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer(mock(ConnectionFactory.class));
+		container.setQueueNames("foo");
+		container.setRecoveryBackOff(new FixedBackOff(100, 3));
+		container.setConcurrentConsumers(10);
+		container.afterPropertiesSet();
+		container.start();
```

Comment:
This needs some polishing to avoid

> 2015-05-07 11:20:11,855 WARN SimpleAsyncTaskExecutor-1 [org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer] - <Consumer raised exception, processing can restart if the connection factory supports it>
java.lang.IllegalArgumentException: Connection must not be null